site/learn/expert-advice/security/breached-password-detection.md

```diff
@@ -0,0 +1,155 @@
+
+Have I Been Pwned is one of the good guys. They are among the services sharing this data in an effort to help people. Users can find out if their credentials have been compromised by visiting that site, and people building and operating applications can access the data to help secure their systems. 
+
+There are other actors out there with, shall we say, less hospitable intentions, using breached datasets to gain unauthorized access to systems.
+
+Of course, a great way to avoid unauthorized access is to secure your own systems and encourage good security practices such as strong passwords and two factor authentication for your application users. Unfortunately, it's common for people to reuse passwords across different applications and services. No matter how good your security is, if credentials are shared between systems, your application is as vulnerable as they are, should they suffer a data breach. Wikipedia has a [list of hundreds of data breaches](https://en.wikipedia.org/wiki/List_of_data_breaches), including some from very large organizations.
+
+## What is breached password detection?
+
+Breached password detection can help. This practice consists of the following:
+
+* Finding breached and otherwise compromised lists of passwords. 
+* Optionally building a system to download, process and store these datasets.
+* Configuring or building a system to check passwords when authentication, registration and password events occur.
+* Taking action when a leaked credential is found.
+
+Lets cover each of these in turn, but first, let's discuss some other reasons why you might be interested in implementing this functionality. 
+
+## Why implement breached password detection?
+
+Beyond preventing unauthorized access to your systems, which is of course a substantial benefit desired by most, detecting when a user has compromised credentials has other benefits.
+
+It is a service to your end users. Informing them that their account with you, and any other accounts that may share the same password, has been compromised is useful information. Depending on how you implement the notice, it can be difficult to ignore. It's easy to forget about an email or letter, but harder to ignore being unable to log in to an application you use.
+
+Compromised password detection is also a protective measure that you, as an application developer or operator, can apply without requiring any user action. Until a leaked or insecure password is found, of course. Following the principles of defense in depth, there are other user account security practices that you should evaluate, such as two factor authentication. Enabling many of these require end user cooperation. Secure two factor authentication, for example, requires a relatively sophisticated user who can install and manage a TOTP application, as SMS is not a secure method for authentication.
```

Comment:
Let's add a link proving "as SMS is not a secure method for authentication." There are a bunch of articles and proof of this, we just need to back it up.